zerver/tests/test_message_send.py

```diff
@@ -1518,6 +1518,49 @@ def test_edit_schedule_message(self) -> None:
         self.assertEqual(message.scheduled_timestamp, convert_to_UTC(new_defer_until))
         self.assertEqual(message.delivery_type, ScheduledMessage.SEND_LATER)
 
+    def test_fetch_scheduled_messages(self) -> None:
+        self.login("hamlet")
+        # No scheduled message
+        result = self.client_get("/json/scheduled_messages")
+        self.assert_json_success(result)
+        self.assert_length(orjson.loads(result.content)["scheduled_messages"], 0)
+
+        content = "Test message"
+        defer_until = timezone_now().replace(tzinfo=None) + datetime.timedelta(days=1)
+        defer_until_str = str(defer_until)
+        self.do_schedule_message("stream", "Verona", content + " 1", defer_until_str)
+
+        # Single scheduled message
+        result = self.client_get("/json/scheduled_messages")
+        self.assert_json_success(result)
+        self.assert_length(orjson.loads(result.content)["scheduled_messages"], 1)
+
+        othello = self.example_user("othello")
+        result = self.do_schedule_message(
+            "private", [othello.email], content + " 3", defer_until_str
+        )
+
+        # Multiple scheduled messages
+        result = self.client_get("/json/scheduled_messages")
+        self.assert_json_success(result)
+        self.assert_length(orjson.loads(result.content)["scheduled_messages"], 2)
+
+    def test_delete_scheduled_messages(self) -> None:
+        self.login("hamlet")
+
+        content = "Test message"
+        defer_until = timezone_now().replace(tzinfo=None) + datetime.timedelta(days=1)
+        defer_until_str = str(defer_until)
+        self.do_schedule_message("stream", "Verona", content, defer_until_str)
+        message = self.last_scheduled_message()
+
+        result = self.client_delete(f"/json/scheduled_messages/{message.id}")
```

Comment:
Can you do a `self.api_delete` test to have another user try to delete it first, and confirm that returns an error and doesn't delete it? It's important to have this sort of access control / security logic in the first version of this sort of thing.

Reply:
Okay, done.